capstone-app/src/app/modules/photos/components/photos-section/photos-section.component.ts

```diff
@@ -1,17 +1,111 @@
-import { Component } from '@angular/core';
+import {
+  Component,
+  Input,
+  OnChanges,
+  ChangeDetectorRef,
+  AfterViewChecked,
+} from '@angular/core';
 import { FormControl, Validators } from '@angular/forms';
+import MockPlacesPa from 'src/app/modules/photos/assets/mock-places-phoenixville.json';
+import MockPlacesBos from 'src/app/modules/photos/assets/mock-places-boston.json';
+import { NavItem } from '../../nav-item';
 
 @Component({
   selector: 'app-photos-section',
   templateUrl: './photos-section.component.html',
-  styleUrls: ['./photos-section.component.scss']
+  styleUrls: ['./photos-section.component.scss'],
 })
-export class PhotosSectionComponent {
+export class PhotosSectionComponent implements OnChanges, AfterViewChecked {
   limitControl = new FormControl(10, [Validators.min(1), Validators.max(10)]);
   maxPhotos = 10;
+  mockPlacesPa: any = (MockPlacesPa as any).places;
+  mockPlacesBos: any = (MockPlacesBos as any).places;
+  mockPlaces: any;
+  /**
+   * These variables are meant to detect the categories markers belong to.
+   * uniqueTypes: Some markers belong to more than one category.
+   * allTypes: The types before extracting the unique ones
+   */
+  uniqueTypes: string[];
+  allTypes: string[];
+  navItems: NavItem[] = [];
+  markerFilter = '';
+  @Input() city: string;
 
-  updateLimit(newLimit: number) {
-    this.maxPhotos = newLimit;
-    this.limitControl = new FormControl(newLimit);
+  constructor(private cdr: ChangeDetectorRef) {}
+
+  // Until the photos section is connected to the maps, only mocks are supported
+  ngOnChanges() {
+    if (this.city.includes('Boston')) {
+      this.mockPlaces = this.mockPlacesBos;
+    } else {
+      this.mockPlaces = this.mockPlacesPa;
+    }
+    this.extractUniqueTypes();
+    this.populateNavItems();
+  }
+
+  ngAfterViewChecked() {
+    this.cdr.detectChanges();
+  }
+
+  /**
+   * Organzies marker input data into object compatible with menu.
+   */
+  populateNavItems() {
```

Comment:
It seems like this component should have more tests to accommodate all of the changes. What particularly stands out to me is a test that the nav types that show up in the menu are the ones that are expected.